src/mctpd.c

```diff
@@ -3512,6 +3901,41 @@ static int add_net(ctx *ctx, int net)
 	return 0;
 }
 
+static int add_interface(ctx *ctx, int ifindex)
+{
+	link_userdata *lmUserData = (link_userdata *)malloc(sizeof(link_userdata));
```

Comment:
No cast needed.

Also, you don't seem to be `free()`ing this. You'll need to do so on the `MCTP_NL_DEL_LINK` notification.